src/vaadin-form-layout.html

```diff
@@ -470,35 +463,40 @@
           }
 
           let col = 0;
-          Array.from(this.children).forEach((child) => {
-            if (getComputedStyle(child).display !== 'none') {
-              let colspan;
-
+          Array.from(this.children)
+            .filter(child => child.localName === 'br' || getComputedStyle(child).display !== 'none')
+            .forEach((child, index, children) => {
               if (child.localName === 'br') {
-                colspan = this._columnCount - col;
-              } else {
-                colspan = this._naturalNumberOrOne(parseFloat(child.getAttribute('colspan')));
-
-                // Never span further than the number of columns
-                colspan = Math.min(colspan, this._columnCount);
+                // Reset column count on line break
+                col = 0;
+                return;
               }
 
+              let colspan;
+              colspan = this._naturalNumberOrOne(parseFloat(child.getAttribute('colspan')));
+
+              // Never span further than the number of columns
+              colspan = Math.min(colspan, this._columnCount);
+
               const childRatio = colspan / this._columnCount;
 
-              if (child.localName === 'br') {
-                // IE and Edge only accept fixed padding
-                child.style.paddingRight = `calc(${childRatio * containerWidth}px - ${1 - childRatio} * ${columnSpacing})`;
-              } else {
-                // Note: using 99.9% for 100% fixes rounding errors in MS Edge
-                // (< v16), otherwise the items might wrap, resizing is wobbly.
-                child.style.width = `calc(${childRatio * 99.9}% - ${1 - childRatio} * ${columnSpacing})`;
-              }
+              // Note: using 99.9% for 100% fixes rounding errors in MS Edge
+              // (< v16), otherwise the items might wrap, resizing is wobbly.
+              child.style.width = `calc(${childRatio * 99.9}% - ${1 - childRatio} * ${columnSpacing})`;
 
               if (col + colspan > this._columnCount) {
                 // Too big to fit on this row, let’s wrap it
                 col = 0;
               }
 
+              const nextIndex = index + 1;
+              const nextLineBreak = nextIndex < children.length && children[nextIndex].localName === 'br';
+
+              if (nextLineBreak) {
```

Comment:
BTW maybe combine that with the if-else block below `// At the end edge`? Would be easer to track all the cases if `marginEndProp` is set/removed in the same place.

Reply:
Combined now.